charts/kubernetes-agent/README.md

```diff
@@ -27,28 +27,40 @@ A Helm chart for the Octopus Kubernetes Agent
 | agent.bearerToken | string | `""` | A JWT bearer token use to authenticate with the target Octopus Server |
 | agent.certificate | string | `""` | A base64 formatted x509 certificate used to setup a trust between the agent and target Octopus Server |
 | agent.debug.disableAutoPodCleanup | bool | `false` | Disables automatic pod cleanup |
-| agent.defaultNamespace | string | `""` | The default Kubernetes namespace for deployments |
 | agent.enableMetricsCapture | bool | `true` | True if events should be scraped and added to the metrics config map |
 | agent.image | object | `{"pullPolicy":"IfNotPresent","repository":"octopusdeploy/kubernetes-agent-tentacle","tag":"8.1.1758"}` | The repository, pullPolicy & tag to use for the agent image |
 | agent.logLevel | string | `"Info"` | The log level of the agent. Logs are written to the pod logs as well as to file |
 | agent.machinePolicyName | string | `""` | The machine policy to register the agent with |
 | agent.metadata | object | `{"annotations":{},"labels":{}}` | Additional metadata to add to the agent pod & container |
+| agent.name | string | `""` | The name of the agent |
 | agent.pollingConnectionCount | int | `5` | The number of polling TCP connections to open with the target Octopus Server |
+| agent.registerAs | string | `""` | Can be `DeploymentTarget` or `Worker`. |
```

Comment:
This I'm not sure about, especially if we end up offering the ability to register as both 🤔

Reply:
Yea that's what I was not sure about https://github.com/OctopusDeploy/helm-charts/pull/184#discussion_r1645396933

The challenge of having a permissive structure now is how we convey that it can't be both. Perhaps with extra validation on the tentacle `configure-and-run.sh` script side?

Reply:
Actually the more I think about it.. we should leave the option open to have both in the Helm Chart, and it's probably sufficient to have validation on the Tentacle start script which will return an error message for now if both are set.

Reply:
In which case I prefer to have an `.isRegistered` option on each of the worker and deploymentTarget levels.

Reply:
There is kinda prior art for having a property with "option 1", "option 2", "option 1 or option 2" in the `tenantedDeploymentParticipation` property. But in our case it would be "option 1 AND option2" I guess. 🤔

Reply:
The one thing I don't like about the `.isRegistered` property is it does sound like you could change it later and it will update the registration. Maybe `.registerOnInitialisation` or `.registerAsWorker`.